From 8da4a2c9e4216b04081f44b56cd179a496fa951e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 18:39:13 +0300
Subject: [PATCH] Revert "[SW-77251] common: add error apis"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit dede6a70766b89a0431e52bd031c9b0f15c911b8.

Change-Id: I12c90a4c7e102b072ef27302658658dc024f114b
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index e538eb9..6f13d1c 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -16,16 +16,6 @@
 #include "boot_desc.h"
 #include <common.h>
 
-struct hl_rsp_err {
-	union {
-		struct {
-			uint32_t major;
-			uint32_t minor;
-		};
-		uint64_t error;
-	};
-};
-
 struct arcboot_global_s arcboot_rom_global = {
 	.config_fuse = 0,
 	.config2_fuse = 0,
@@ -58,37 +48,6 @@
 	.pci_link_up = false,
 	.tpm_validation_pass = false,
 };
-
-static struct hl_rsp_err pending_error;
-
-bool arcboot_error_inform(uint32_t major, uint32_t minor)
-{
-	/*
-	 * ensuring sending the first error occurrence
-	 * to reset the error just use the clr_on_read
-	 */
-	if (!pending_error.error) {
-		pending_error.major = major;
-		pending_error.minor = minor;
-		return true;
-	}
-
-	return false;
-}
-
-bool arcboot_error_is_pending(void)
-{
-	return !!pending_error.error;
-}
-
-uint32_t arcboot_error_read_and_clr(uint32_t *major, uint32_t *minor)
-{
-	*major = pending_error.major;
-	*minor = pending_error.minor;
-
-	pending_error.error = 0;
-	return *major;
-}
 
 #define ARCBOOT_STATUS_REG_SZ_BITS		32
 __weak void arcboot_rom_status_update(enum arcboot_rom_progress progress)
diff --git a/include/arcboot.h b/include/arcboot.h
index 584cb9a..9f8f5c7 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -269,9 +269,6 @@
 void arcboot_rom_load_recovery(void);
 int arcboot_rom_mem_init(uint64_t addr, uint32_t size);
 void arcboot_rom_verify(void);
-uint32_t arcboot_error_read_and_clr(uint32_t *major, uint32_t *minor);
-bool arcboot_error_inform(uint32_t major, uint32_t minor);
-bool arcboot_error_is_pending(void);
 
 #define CLEAR_DMA_SECURITY_PROGRESS 0x3
 
